src/library/scala/collection/MapView.scala

```diff
@@ -72,13 +106,42 @@ object MapView {
   }
 
   @SerialVersionUID(3L)
-  private class MapViewMapFactory[K, V] extends MapFactory[({ type l[X, Y] = View[(X, Y)] })#l] {
-    def newBuilder[X, Y]: Builder[(X, Y), View[(X, Y)]] = View.newBuilder[(X, Y)]
-    def empty[X, Y]: View[(X, Y)] = View.empty
-    def from[X, Y](it: IterableOnce[(X, Y)]): View[(X, Y)] = View.from(it)
+  class Filter[K, +V](underlying: SomeMapOps[K, V], isFlipped: Boolean, p: ((K, V)) => Boolean) extends AbstractMapView[K, V] {
+    def iterator: Iterator[(K, V)] = underlying.iterator.filterImpl(p, isFlipped)
+    def get(key: K): Option[V] = underlying.get(key) match {
```

Comment:
This is slightly inconsistent with the concat `get` which uses the higher-order `orElse` instead of a pattern match.  Can you either use `filter` here or pattern-match the other one?  (I guess pattern match the other is better for performance.)

Reply:
Done